cache/elasticache.tf

```diff
@@ -0,0 +1,25 @@
+locals {
+  azs = [
+    "${var.aws_region}a",
+    "${var.aws_region}b",
+    "${var.aws_region}c"
+  ]
+}
+
+resource "aws_elasticache_replication_group" "cache" {
+  replication_group_id          = "fp-${var.fp_context}-cache"
+  automatic_failover_enabled    = var.fp_context == "production"
+  availability_zones            = var.fp_context == "production" ? local.azs : [local.azs[0]]
```

Comment:
I think most regions have min 3 availability zones so we should be fine.  If we want to query data component, we can do something like "${data.aws_availability_zones.xxx.names[count.index]}"

Not a major issue.

Reply:
The production environment will have more than one node in the cluster so it can take advantage of high availability with each node being in a different AZ. But for non-production environments, we will only have one node to cut down on costs, which is why we are specifying only one AZ.